config/source.go

```diff
@@ -55,23 +57,29 @@ func ParseSource(cfg map[string]string) (Source, error) {
 	sourceConfig := Source{
 		Configuration:  config,
 		OrderingColumn: strings.ToUpper(cfg[OrderingColumn]),
-		KeyColumn:      strings.ToUpper(cfg[KeyColumn]),
 		BatchSize:      defaultBatchSize,
 	}
 
-	if cfg[Columns] != "" {
-		columnsSl := strings.Split(cfg[Columns], ",")
+	if cfg[KeyColumns] != "" {
+		keyColumns := strings.Split(strings.ReplaceAll(cfg[KeyColumns], " ", ""), ",")
+		for i := range keyColumns {
+			if keyColumns[i] == "" {
+				return Source{}, fmt.Errorf("invalid %q", KeyColumns)
+			}
 
-		// converts columns to uppercase
-		for i := range columnsSl {
-			columnsSl[i] = strings.TrimSpace(strings.ToUpper(columnsSl[i]))
+			sourceConfig.KeyColumns = append(sourceConfig.KeyColumns, strings.ToUpper(keyColumns[i]))
 		}
+	}
 
-		if err = validateColumns(sourceConfig.OrderingColumn, sourceConfig.KeyColumn, columnsSl); err != nil {
-			return Source{}, err
-		}
+	if cfg[Columns] != "" {
+		columnsSl := strings.Split(strings.ReplaceAll(cfg[Columns], " ", ""), ",")
+		for i := range columnsSl {
+			if columnsSl[i] == "" {
+				return Source{}, fmt.Errorf("invalid %q", Columns)
```

Comment:
we can just skip it using continue and add sdk.logger warning, what do you think about it?

Reply:
We've decided that if the user specifies `id, ,name` value, the system should return an error.